### DISCUSSION TITLE
Adding sink and recombination to the Multi-phase poly-crystalline Allen-Cahn Kernel

### DISCUSSION THREAD

**Discussion by jackeggemeyer** [idaholab/moose#27285] [Q&A Modules: Phase field] [answered]
I am trying to add sink and recombination to the Multi-phase poly-crystalline Allen-Cahn Kernel. The top equation is the equation already used by MOOSE. The second equation is from: https://www.sciencedirect.com/science/article/pii/S0927025622004839#! and it adds vacancy production, vacancy-interstial recombination, and sink terms to the grand potential equation in terms of chemical potential. The "MaskedBodyForce" kernal is being used to create the vacancy production part h_m*s^o_v. Does moose have kernals that could be used to simulate the two other parts of the equation? I am new to MOOSE.
![image](https://github.com/idaholab/moose/assets/160534018/310a01e5-8dc3-4661-b79e-9e2be9055c53)

**Comment by richmondodufisan** [comment, marked as answer]
I don't think so, you'll probably have to create your own custom kernel. it will work seamlessly with the phase field module. 

https://mooseframework.inl.gov/getting_started/examples_and_tutorials/examples/ex02_kernel.html

this was the video I watched to learn all of MOOSE's features a while back, it's 12 hrs long (took me a week) but it really is worth it:

https://www.youtube.com/watch?v=2tJwBsYaLaI&list=PLHI-fE3lo53508O5k9nli7BNMNdxFQSF5&index=1&t=11789s
